writer_test.go

```diff
@@ -25,6 +26,110 @@ func TestWriteAndRetryFails(t *testing.T) {
 	}
 }
 
+func TestWriteFormatters(t *testing.T) {
+	tests := []struct {
+		f Formatter
+	}{
+		{nil},
+		{UnixFormatter},
+		{RFC3164Formatter},
+		{RFC5424Formatter},
+		{DefaultFormatter},
+	}
+
+	for _, test := range tests {
+		done := make(chan string)
+		addr, sock, srvWG := startServer("udp", "", done)
+		defer sock.Close()
+		defer srvWG.Wait()
+
+		w := Writer{
+			priority: LOG_ERR,
+			tag:      "tag",
+			hostname: "hostname",
+			network:  "udp",
+			raddr:    addr,
+		}
+
+		w.Lock()
+		err := w.connect()
+		if err != nil {
+			t.Errorf("failed to connect: %v", err)
+			w.Unlock()
+		}
+		w.Unlock()
+		defer w.Close()
+
+		w.SetFormatter(test.f)
+
+		f := test.f
+		if f == nil {
+			f = DefaultFormatter
+		}
+		expected := strings.TrimSpace(f(LOG_ERR, "hostname", "tag", "this is a test message"))
+
+		_, err = w.Write([]byte("this is a test message"))
+		if err != nil {
+			t.Errorf("failed to write: %v", err)
+		}
+		sent := strings.TrimSpace(<-done)
+		if sent != expected {
+			t.Errorf("expected to use the formatter, got %v, expected %v", sent, expected)
```

Comment:
I don't understand what this error message means. This shows up when the result of formatting a string using a specific formatter does not match an expectation. Would a more descriptive error message/assertion be "expected to use formatter <formatter-name>, got <result>, expected <expecation`"? 

Is this is asserting that the function returned used the wrong formatter or that it returned the wrong string (or both by the fact the wrong formatter produced a bad string).